### PR TITLE
Beds 152/archived dashboards

### PR DESCRIPTION
**Pull request by Eisei24**

### CODE REVIEW

File: backend/pkg/api/enums/validator_dashboard_enums.go [hunk 1]
```diff
@@ -270,6 +270,66 @@ var VDBManageValidatorsColumns = struct {
 	VDBManageValidatorsWithdrawalCredential,
 }
 
+// ----------------
+// Validator Dashboard Archived Reasons
+
+type VDBArchivedReason int
+
+var _ EnumFactory[VDBArchivedReason] = VDBArchivedReason(0)
```
Comment (remoterami) [inline backend/pkg/api/enums/validator_dashboard_enums.go:278]: Does this need to be an `EnumFactory`? If we're not validating the DB value in https://github.com/gobitfly/beaconchain/pull/692/files#diff-600151bf2b435c255348f0c1a82d41fcde5c86a5a0fec8040be3b6eb5ff1de1fR100, I don't think `NewFromString()` is going to be used anywhere else

**Review by remoterami — CHANGES_REQUESTED** [review]
mostly looks good, some minor comments

File: backend/pkg/api/data_access/user.go [hunk 2]
```diff
@@ -547,53 +548,107 @@ func (d *DataAccessService) GetFreeTierPerks(ctx context.Context) (*t.PremiumPer
...
+			return err
+		}
+
+		for _, row := range dbReturn {
+			entry := DashboardCount{
+				Id:             row.Id,
+				GroupCount:     row.GroupCount,
+				ValidatorCount: row.ValidatorCount,
+			}
+			validatorDashboardCountMap[row.Id] = entry
```
[the remark this answers is not in the document]
Comment (Eisei24) [reply backend/pkg/api/data_access/user.go:636]: Right, done.

File: backend/pkg/api/enums/validator_dashboard_enums.go (hunk 1 above)
Comment (Eisei24) [reply backend/pkg/api/enums/validator_dashboard_enums.go:278]: After trying to understand more how `EnumFactory` is used I agree and changed it to the normal `Enum`.

File: backend/pkg/commons/db/migrations/postgres/TBD_is_archived.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
+-- +goose Up
+-- +goose StatementBegin
+SELECT 'up SQL query - add is_archived column';
+ALTER TABLE users_val_dashboards ADD COLUMN IF NOT EXISTS is_archived TEXT;
```
[the remark this answers is not in the document]
Comment (Eisei24) [reply backend/pkg/commons/db/migrations/postgres/TBD_is_archived.sql:4]: As discussed we will leave it as `is_archived`.

**Review by remoterami — APPROVED** [review]
LGTM

**Pull request merged**